src/components/PageDefinitions.js

```diff
@@ -296,38 +329,38 @@ export class PageDefinitions extends AbstractPageDefinitions {
     )
   }
 
+  async onGistDrop(urlString) {
```

Comment:
As for the above comment, i would likely remove any kind of specific provider management from a single component.
These could be reusable methods for other pages or lists (like the one in the new Browse page) where we probably need to share them.

A utils class that manage all these different implementation it's the best way imo

Reply:
I'd like to have a drop component that can be configured with handlers. The current direction is towards that where the handlers just come in as an array and the drop handler searches for the appropriate handler and runs it. There will be future refinements to be sure.